File: typescript/lld/booking/movie-ticket/Booking.ts
export enum BookingStatus { PENDING, CONFIRMED, CANCELLED }

export class Booking {
    status = BookingStatus.PENDING;
    constructor(
        public readonly id: string,
        public readonly userId: string,
        public readonly showId: string,
        public readonly seatIds: string[],
        public readonly amount: number,
    ) {}
}

File: java/lld/booking/movie-ticket/Booking.java
import java.util.List;

public class Booking {
    public final String id, userId, showId;
    public final List<String> seatIds;
    public final double amount;
    public BookingStatus status = BookingStatus.PENDING;

    public Booking(String id, String userId, String showId,
                   List<String> seatIds, double amount) {
        this.id = id; this.userId = userId; this.showId = showId;
        this.seatIds = seatIds; this.amount = amount;
    }
}